### PR TITLE
Lesson03 Labs & Mailroom + Lesson04 Mailroom & Trigrams

### PR DESCRIPTION
**Pull request by nicksama88**
Completed first run-through of the Lesson03 labs.

### CODE REVIEW

**Comment by nicksama88** [comment]
Went through the exercises for dict_lab.py

File: students/Nick_Haury/Lesson03/list_lab.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+# Series 1
+
+print("Series 1")
+fruits = ["Apples", "Pears", "Oranges", "Peaches"]
+print(fruits)
+new_fruit = input("Please enter a fruit to add to the end of the list: ")
+fruits.append(new_fruit)
+print(fruits)
+fruit_index = int(input("Please enter the number place of a fruit in the list: "))
```
Comment (charlesrobison) [inline students/Nick_Haury/Lesson03/list_lab.py:16]: Recommend len() here to show the number of items in the list.

File: students/Nick_Haury/Lesson03/list_lab.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+print(str(fruit_index) + ": " + fruits[fruit_index - 1])
+fruits = ["Mangoes"] + fruits
+print(fruits)
+fruits.insert(0, "Blueberries")
+print(fruits)
+temp_fruits = []
+for fruit in fruits:
+    if fruit.lower()[0] == "p":
```
Comment (charlesrobison) [inline students/Nick_Haury/Lesson03/list_lab.py:24]: Nice use of lower().

File: students/Nick_Haury/Lesson03/list_lab.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+print("")
+print("Series 2")
+print(fruits)
+fruits2 = fruits[:]
+fruits2.pop()
+print(fruits)
+found_fruit = False
+while not found_fruit:
+    fruit_to_remove = input("what fruit (case sensitive) should be removed?: ")
```
Comment (charlesrobison) [inline students/Nick_Haury/Lesson03/list_lab.py:38]: You can force input to be capitalized too, if you want.

File: students/Nick_Haury/Lesson03/list_lab.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+fruits_copy = fruits[:]
+for fruit in fruits_copy:
+    yes_no_input = ""
+    while yes_no_input not in ("yes", "no"):
+        yes_no_input = input("Do you like " + fruit.lower() + "?: ")
+        if yes_no_input not in ("yes", "no"):
+            print("answer must be 'yes' or 'no'")
+        elif yes_no_input == "no":
+            fruits.remove(fruit)
+print(fruits)
```
Comment (charlesrobison) [inline students/Nick_Haury/Lesson03/list_lab.py:61]: Good!

File: students/Nick_Haury/Lesson03/strformat_lab.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+print("Task Six")
+name_age_cost_list = [["name1", 1, 12.50], ["name2", 2, 500.00], ["name3", 50, 100000.00]]
+header_list = ["Name", "Age", "Cost"]
+print(f"|{header_list[0]:^10}|{header_list[1]:^5}|{header_list[2]:^16}|")
+for item in name_age_cost_list:
+    print(f"|{item[0]:10}|{item[1]:5}| ${item[2]:14,.2f}|")
+print()
+ten_int_tup = (1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
+print(("{:5}\n"*len(ten_int_tup)).format(*ten_int_tup))
```
Comment (charlesrobison) [inline students/Nick_Haury/Lesson03/strformat_lab.py:65]: Works well and looks good - nice job!

File: students/Nick_Haury/Lesson04/mailroom.py [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+        with open(f'{key}.txt', 'w') as f:
+            f.write(print_email(key, donors[key][-1]))
+    print()
+
+if __name__ == "__main__":
+    '''
+    run main prompt
+    '''
+
+    main()
```
Comment (charlesrobison) [inline students/Nick_Haury/Lesson04/mailroom.py:150]: Great!

File: students/Nick_Haury/Lesson04/trigrams.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        print("You must pass in a filename argument in the form of "
+        "'filename.txt' along with an int argument for the number of pages "
+        "you'd like the created text to be.  ex: trigrams.py text.txt 100")
+        sys.exit(1)
+    words = read_file(file_name)
+    trigrams = build_trigrams(words.split())
+    text = build_text(trigrams, int(length))
+    print(text)
+    # for k, v in trigrams.items():
+    #     print(k, v)
```
Comment (charlesrobison) [inline students/Nick_Haury/Lesson04/trigrams.py:63]: Excellent!

**Pull request merged**